### PR TITLE
docs(features): add plugin-autostart

### LINKED ISSUE #1474: Create Autostart guide

### PR DESCRIPTION
**Pull request by pewsheen**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- What does this PR change? Give us a brief description. 
Updates `/features/autostart/`

### CODE REVIEW

File: src/content/docs/features/autostart.mdx [hunk 1]
```diff
@@ -3,12 +3,136 @@ title: Autostart
...
+
+By default all plugin commands are blocked and cannot be accessed.
+You must define a list of permissions in your `capabilities` configuration.
+
+See [Access Control List](/references/v2/acl) for more information.
+
+`src-tauri/capabilities/main.json`
+
+```json
+{
```
Comment (vasfvitor) [inline src/content/docs/features/autostart.mdx]:
```suggestion
```json title="src-tauri/capabilities/main.json"
{
```
reference: https://expressive-code.com/key-features/frames/#usage-in-markdown--mdx

File: src/content/docs/features/autostart.mdx [hunk 2]
```diff
@@ -3,12 +3,136 @@ title: Autostart
...
+
+  </TabItem>
+  <TabItem label="Manual">
+
+  1. Run `cargo add tauri-plugin-autostart` to add the plugin to the project's dependencies in `Cargo.toml`.
+
+  2. Modify `lib.rs` to initialize the plugin:
+
+  ```rust
+  // lib.rs
```
Comment (vasfvitor) [inline src/content/docs/features/autostart.mdx]:
```suggestion
  ```rust title="lib.rs" ins={1, 6}
```

Expressive Code supports comments as file title, but I suggest to be specific and utilize title.
`ins={1, 6}` will highlight the 6th and 1st line

The other plugins guides still have the comment syntax but this was before Expressive Code support in Starlight. They're all pending a little touch up, including with the new permission stuff and beta release

**Review by vasfvitor — COMMENTED** [review]
Few notes about Expressive Code

**Comment by pewsheen** [comment]
> Few notes about Expressive Code

TIL, thanks for your explanation!

**Comment by vasfvitor** [comment]
I should've linked the starlight docs to expressive code tho https://starlight.astro.build/guides/authoring-content/#expressive-code-feature instead of es docs, for simplicity

File: src/content/docs/features/autostart.mdx [hunk 3]
```diff
@@ -3,12 +3,133 @@ title: Autostart
...
 
 <PluginLinks plugin="autostart" />
 
-<Stub>
-	Based on
-	https://github.com/tauri-apps/plugins-workspace/tree/v2/plugins/autostart
-</Stub>
+Automatically launch your application at system startup.
+
+Supports Windows, Mac (via AppleScript or Launch Agent), and Linux.
```
Comment (vasfvitor) [inline src/content/docs/features/autostart.mdx]: For consistency with other guides
```suggestion
## Supported Platforms

- Windows
- Mac (via AppleScript or Launch Agent)
- Linux
```

**Review by vasfvitor — APPROVED** [review]

**Pull request merged**